File: java/liskov-plus-refactor/src/main/java/no/bouvet/solid/lsp/services/PaymentFactory.java
package no.bouvet.solid.lsp.services;

import no.bouvet.solid.lsp.domain.PayPalPayment;
import no.bouvet.solid.lsp.domain.Payment;
import no.bouvet.solid.lsp.domain.WorldPayPayment;

public class PaymentFactory {

    public static Payment getPayment(PaymentType paymentServiceType) {

        switch (paymentServiceType) {
        case PayPal:
            return new PayPalPayment("Andras", "Passw0rd");
        case WorldPay:
            return new WorldPayPayment("Andras", "Passw0rd", "ABC");
        default:
            throw new UnsupportedOperationException("No such service");
        }

    }

}

File: dotnet/Solid/RefactoredSolid/LSP/Example2/Services/PaymentFactory.cs
using System;
using RefactoredSolid.LSP.Example2.Domain;

namespace RefactoredSolid.LSP.Example2.Services
{
    public class PaymentFactory
    {
        public static Payment GetPayment(PaymentType paymentServiceType)
        {
            switch (paymentServiceType)
            {
                case PaymentType.PayPal:
                    return new PayPalPayment("...", "Passw0rd");
                case PaymentType.WorldPay:
                    return new WorldPayPayment("...", "Passw0rd", "ABC");
                default:
                    throw new NotSupportedException("No such service");
            }
        }
    }
}
